File: app/src/main/java/com/admin/theapp/utils/Decoder.java
package com.admin.theapp.utils;

import android.content.Context;
import android.graphics.BitmapFactory;
import android.graphics.drawable.BitmapDrawable;
import android.support.annotation.NonNull;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public final class Decoder {

    @NonNull
    private final Context context;

    @Inject
    public Decoder(@NonNull Context context) {
        this.context = context;
    }

    public BitmapDrawable decode(final byte[] data) {
        return new BitmapDrawable(context.getResources(), BitmapFactory.decodeByteArray(data, 0, data.length));
    }
}

File: app/src/main/java/com/admin/theapp/utils/Decoder.kt
package com.admin.theapp.utils

import android.content.Context
import android.graphics.BitmapFactory
import android.graphics.drawable.BitmapDrawable
import java.io.InputStream

import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class Decoder @Inject constructor(private val context: Context) {

    fun decode(data: ByteArray) =
            BitmapDrawable(context.resources, BitmapFactory.decodeByteArray(data, 0, data.size))

    fun decode(data: InputStream) =
            BitmapFactory.decodeStream(data)
}
